core/src/main/java/com/linecorp/armeria/client/HttpClientBuilder.java

```diff
@@ -56,6 +64,17 @@ public HttpClientBuilder(URI uri) {
         this.uri = URI.create(SerializationFormat.NONE + "+" + uri);
     }
 
+    /**
+     * Creates a new instance.
+     *
+     * @throws IllegalArgumentException if the {@link Scheme} is not one of the fields
+     *                                  in {@link SessionProtocol}
+     */
+    public HttpClientBuilder(Scheme scheme, Endpoint endpoint) {
+        this.scheme = requireNonNull(scheme, "scheme");
```

Comment:
After converting a `SessionProtocol` to a `Scheme` by adding `none+`, we still should validate the scheme.